### PR TITLE
Style

### PR DESCRIPTION
**Pull request by txdv**
As discussed in #155

### CODE REVIEW

**Comment by saghul** [comment]
LGTM, but it needs a better commit log, maybe "fs, pipe: ..." and then explaing what functions were changed.

Also, please update the documentation adding some `.. versionchanged:: 1.3.0 the returned length no longer includes the terminating null byte` where appripriate. Last, place some asserts in the tests to verify the new behavior.

Thanks Andrius!

File: src/win/pipe.c [hunk 1]
```diff
@@ -1910,7 +1910,7 @@ int uv_pipe_open(uv_pipe_t* pipe, uv_file file) {
 }
 
 
-int uv_pipe_getsockname(const uv_pipe_t* handle, char* buf, size_t* len) {
+int uv_pipe_getsockname(const uv_pipe_t* handle, char* buffer, size_t* size) {
```
Comment (saghul) [inline src/win/pipe.c]: You need to adapt the function to this rename.

File: src/uv-common.c [hunk 2]
```diff
@@ -359,7 +359,7 @@ int uv_send_buffer_size(uv_handle_t* handle, int *value) {
   return uv__socket_sockopt(handle, SO_SNDBUF, value);
 }
 
-int uv_fs_event_getpath(uv_fs_event_t* handle, char* buf, size_t* len) {
+int uv_fs_event_getpath(uv_fs_event_t* handle, char* buffer, size_t* size) {
```
Comment (saghul) [inline src/uv-common.c]: Adjust the usage to the new names.

**Comment by saghul** [comment]
Ups, I was a bit too quick on the LGTM. @txdv you need to fix the functions to the renamed parameters.

File: src/win/pipe.c (hunk 1 above)
Comment (txdv) [inline src/win/pipe.c]: LOL I must be drunk. I'll do that.

**Comment by txdv** [comment]
The documentation as well probably...

**Comment by txdv** [comment]
The first commit is a separate commit, the rest I will just squash.

**Comment by saghul** [comment]
Code looks fine, please adjust the docs as I mentioned here: https://github.com/libuv/libuv/pull/159#issuecomment-71188808 also place some asserts in related tests to make sure we don't change the behavior in the future.

**Comment by txdv** [comment]
Want me to squash everything into one commit?

**Comment by txdv** [comment]
Looks like the windows part has a trailing whitespace as well.

File: test/test-fs-event.c [hunk 3]
```diff
@@ -642,6 +642,7 @@ TEST_IMPL(fs_event_getpath) {
   len = sizeof buf;
   r = uv_fs_event_getpath(&fs_event, buf, &len);
   ASSERT(r == 0);
+  ASSERT(buf[len - 1] != 0);
```
Comment (saghul) [inline test/test-fs-event.c]: Please also check that strlen(buf) == len.

**Comment by saghul** [comment]
> Want me to squash everything into one commit?
> 
> Or have two? One with behavioral change, the other style?

Please make it 2, the style (naming) and the behavior change.

**Comment by txdv** [comment]
Squashed, pushed. Tested it on windows and linux.

File: src/win/pipe.c [hunk 4]
```diff
@@ -1984,34 +1984,33 @@ int uv_pipe_getsockname(const uv_pipe_t* handle, char* buf, size_t* len) {
...
                                 NULL);
   if (!addrlen) {
-    *len = 0;
+    *size = 0;
     err = uv_translate_sys_error(GetLastError());
     goto error;
-  } else if (pipe_prefix_len + addrlen + 1 > *len) {
+  } else if (pipe_prefix_len + addrlen > *size) {
     /* "\\\\.\\pipe" + name + '\0' */
-    *len = pipe_prefix_len + addrlen + 1;
```
Comment (saghul) [inline src/win/pipe.c]: Please update the comment above this line, we no longer include the \0

File: test/test-fs-event.c (hunk 3 above)
Comment (saghul) [inline test/test-fs-event.c]: Check my coment above: https://github.com/libuv/libuv/pull/159/files#r23524713

**Comment by saghul** [comment]
@txdv almost there, a couple fo nits and it's good to go.

**Comment by txdv** [comment]
The windows buffers are not zeroed out by default and have junk at the after, so strlen(buf) == len doesn't really work unless we put a 0 right after the string ends.

**Comment by txdv** [comment]
Do you really want to have the second commit?

**Comment by saghul** [comment]
> The windows buffers are not zeroed out by default and have junk at the after, so strlen(buf) == len doesn't really work unless we put a 0 right after the string ends.

Ok, drop that then.

**Comment by saghul** [comment]
> Do you really want to have the second commit?

Yep

**Comment by txdv** [comment]
With second commit I mean "add additional strlen check", which puts a '\0' at the end of the string if the buffer is big enough.

Now I am confused, drop or not drop?

**Comment by txdv** [comment]
I think
```c
ASSERT(memcmp(buf, TEST_PIPENAME, len) == 0);
```

is enough, we shouldn't put a 0 terminated string at the end. If the user really wants it to have it like that he can do it himself very easily with buffer[len] = '\0';

**Comment by saghul** [comment]
> I think
> 
> ASSERT(memcmp(buf, TEST_PIPENAME, len) == 0);
> 
> is enough, we shouldn't put a 0 terminated string at the end. If the user really wants it to have it like that he can do it himself very easily with buffer[len] = '\0';

The check is correct. About terminating the string, what you said doesn't hold if the supplied buffer is completely filled though.

**Comment by txdv** [comment]
> The check is correct. About terminating the string, what you said doesn't hold if the supplied buffer is completely filled though.

Em, he can create a buffer with + 1 for the null termination?

**Comment by saghul** [comment]
> Em, he can create a buffer with + 1 for the null termination?

The user doesn't know the size unless it queries it beforehand by passing a NULL buffer, which i"m not 100% sure we support. So the required size is pretty much unknown. It's not a problem if the docs say the strings are **not** NULL terminated though.

**Comment by txdv** [comment]
```
char buffer[1024 + 1];
size_t len;
len = 1024; 
uv_pipe_getsockname(handle, buffer, &len);
buffer[len] = '\0';
```

Anyway, you told me to add the strlen check and I think it is bad, why did you make me add it?

**Comment by saghul** [comment]
> Anyway, you told me to add the strlen check and I think it is bad, why did you make me add it?

Because I forgot about the fact that Windows doesn't null-terminate the buffers, Unix does IIRC. Discard it.

**Comment by txdv** [comment]
Ok, no problem, I just don't want to waste more time on this.

**Comment by saghul** [comment]
CI: https://jenkins-iojs.nodesource.com/view/libuv/job/libuv+any-pr+multi/57/ (We do have some known and unrelated failures.)

**Comment by saghul** [comment]
Thanks, Andrius. Landed in 1e59ab1d49bace6d43726b076d2e820f318c53f9 and 2bfa2e5e22757d5c7252dc7594cf4c36e86b8b58.

**Pull request closed without merging**